src/sqlfluff/api/simple.py

```diff
@@ -0,0 +1,54 @@
+"""The simple public API methods."""
+
+from ..core import Linter
+
+
+def _unfiy_str_or_file(sql):
+    """Unify string and files in the same format."""
+    if not isinstance(sql, str):
+        try:
+            sql = sql.read()
+        except AttributeError:
+            raise TypeError("Value passed as sql is not a string or a readable object.")
+    return sql
+
+
+def lint(sql, dialect="ansi"):

```

Comment:
Should we consider letting the caller specify which rules to check?

Reply:
yes. I'll add that. I assume the choice would be by code reference, so I'll do that first for the next round of review.